atlp-rwanda/ecommerce-app-champions-fe
# #65 chore: Configure ESLint and Git Pre commit hook

## what does this PR do?
- install eslint,prettier,husky dependencies
- configure eslint
- configure prettier
- integrate prettier with eslint
- implement Git pre commit hook
## Descriptions
Configure ESLint and Git Pre commit hook to run tests and ESlint format before commits
## How this should be manually tested:
- clone this repo `git clone https://github.com/atlp-rwanda/ecommerce-app-champions-fe.git`
- checkout to `ch-setup-eslint-pre-commit`
- install the packages by `npm install`
- to test eslint and pre-commit `create file inside src/.  with .js  ` once you done copy code from `App.jsx`  and paste it into your `.js` file 
-  in your terminal  run git command ` git add .` and then try `git commit -m " your message"` you will see how we run lint script to check errors and format script to format our code if any error is found the commit process will be aborted

<img width="1440" alt="Screenshot 2023-04-18 at 22 36 27" src="https://user-images.githubusercontent.com/89464364/232911406-1f5e2cd9-19b7-4898-b176-7b00ca3c64c0.png">

## Trello CARD #65
